src/App.js

```diff
@@ -14,11 +15,124 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends React.Component {
+  state = {
+    startButton: true,
+    reverse: false,
+    sortBy: '',
+    selectLength: 1,
+  };
+
+  reset() {
+    this.setState({
+      reverse: false,
+      sortBy: '',
+      selectLength: 1,
+    });
+  }
+
+  startButton() {
+    this.setState({ startButton: false });
+  }
+
+  goodsRevers() {
+    this.setState(state => ({
+      reverse: !state.reverse,
+    }));
+  }
+
+  goodsSortByLength() {
+    this.setState({ sortBy: 'length' });
+  }
+
+  goodsSortByName() {
+    this.setState({ sortBy: 'name' });
+  }
+
+  selectLength(event) {
+    this.setState({ selectLength: event.target.value });
+  }
+
+  render() {
+    const { startButton, selectLength } = this.state;
+
+    return (
+      <div className="container">
+        {startButton ? (
+          <button
+            type="button"
+            className="container__button"
+            onClick={() => this.startButton()}
+          >
+            Push me:)
+          </button>
+        ) : (
+          <div>
+            <button
+              type="button"
+              className="container__Button-list"
+              onClick={() => this.goodsRevers()}
+            >
+              Reverse
+            </button>
+
+            <button
+              type="button"
+              className="container__Button-list"
+              onClick={() => this.goodsSortByName()}
+            >
+              Sort alphabetically
+            </button>
+
+            <button
+              type="button"
+              className="container__Button-list"
+              onClick={() => this.goodsSortByLength()}
+            >
+              Sort by length
+            </button>
+
+            <button
+              type="button"
+              className="container__Button-list"
+              onClick={() => this.reset()}
+            >
+              Reset
+            </button>
+
+            <div className="container__select">
+              <label htmlFor="minLength" className="container__label">
+                Select the minimum list length:
+                <select
+                  className="container__button-select"
+                  id="minLength"
+                  name="minLength"
+                  onChange={event => this.selectLength(event)}
+                  value={selectLength}
+                >
+                  {[...Array(10).keys()].map((element) => {
+                    const number = element + 1;
+
+                    return (
+                      <option key={number} value={number}>
+                        {number}
+                      </option>
+                    );
+                  })}
+                </select>
+              </label>
+            </div>
+
+            <ListOfGoods
+              goods={goodsFromServer}
+              stateValue={{ ...this.state }}
```

Comment:
if you need state in ListOfGoods maybe move it there? (with buttons and everything?)